### DISCUSSION TITLE
API endpoint for deleting old update status

### DISCUSSION THREAD

**Discussion by vence722** [meilisearch/product#113] [Feedback & Feature Proposal]
**Is your feature request related to a problem? Please describe.**
Our system continuously makes updates to the document, while meilisearch will generate a 'Update' object which helps us to track the asynchronous process. However the update history seems never to be deleted, and currently there's no endpoint for doing it. We want to track the current updating activity by calling the GET /indexes/:index_id/updates, and the response list will grow to thousands of records if our system keeps updating the index.

**Describe the solution you'd like**
We want to have a endpoint to DELETE the past 'Update' object with a specific time range, or number of records to delete.

**Comment by Nick-Mazuk** [comment]
Another potential solution: provide the option for MeiliSearch to delete old updates automatically. For instance, "keep only the last 1,000 updates." Like potentially storing all the updates in an LRU cache.

**Comment by qdequele** [comment]
Hello @Nick-Mazuk @vence722,

Two great features, we already planned to add the ability to delete old updates status. But the idea of a kind of LRU cache is a great idea. We could investigate on that. I invite you both to add a feature request on our [roadmap](https://roadmap.meilisearch.com)! When it will be done, we will close this issue.

**Comment by gmourier** [comment]
Hi @vence722 ! Thanks for this feedback.

Why do you want to remove updates considered as old? Is it a question of performance or information organization/ease of access?

I wonder if adding a filtering capability to the update list endpoint would solve your problem in a simple way if it is a matter of organizing the results to easily find the updates you are interested in.

I ask this because we plan to add filters to the update lists in a near future and I would like to know more about your need! 

Thanks ✌️

**Comment by zehawki** [comment]
I'd like to request the same.

>Is it a question of performance or information organization/ease of access?

It may be neither - perhaps just cleanup.

**Comment by gmourier** [comment]
I'm locking that discussion since it's now possible to:

- Cancel enqueued/processing tasks
- Delete "old" tasks (finished/canceled)

📚[Cancel tasks](https://docs.meilisearch.com/reference/api/tasks.html#cancel-tasks)
📚[Delete tasks](https://docs.meilisearch.com/reference/api/tasks.html#delete-tasks)

Please open a new discussion if you have feedback to share!
